main.go

```diff
@@ -255,13 +262,12 @@ func main() {
 	}
 
 	// Only start the controller when the leader election is won
-	mgr.Add(manager.RunnableFunc(func(stop <-chan struct{}) error {
+	mgr.Add(manager.RunnableFunc(func(ctx context.Context) error {
 		setupLog.Info("starting custom controller")
 
 		// Start the custom controller
-		customController.StartController(stop)
+		customController.StartController(ctx.Done())
 		// If the manager is stopped, signal the controller to stop as well.
-		<-stop
```

Comment:
`<-stop` this should be kept here. otherwise, it will log "stopping the controller" immediately